facilitator/src/task/sqs.rs

```diff
@@ -53,6 +49,7 @@ impl<T: Task> TaskQueue<T> for AwsSqsTaskQueue<T> {
         );
 
         let client = self.sqs_client()?;
+        let runtime = basic_runtime()?;
```

Comment:
[`tokio::runtime::Runtime`](https://docs.rs/tokio/1.3.0/tokio/runtime/struct.Runtime.html) does not implement `std::clone::Clone`, and so a struct which wants to derive `Clone` can't have a `Runtime` member. There is [`tokio::runtime::Handle`](https://tokio-rs.github.io/tokio/doc/tokio/runtime/struct.Handle.html) that can be cloned but I'm pretty sure we don't want that: `basic_runtime` uses [`tokio::runtime::Builder::new_current_thread`](https://docs.rs/tokio/1.3.0/tokio/runtime/struct.Builder.html#method.new_current_thread) so that we can synchronously run async tasks.

Suppose I start on the main thread of a process and then do:

```
let runtime = tokio::runtime::Builder::new_current_thread().build();
let handle_clone = runtime.handle().clone();
thread::spawn(move || {
    // now on thread B
    handle_clone.block_on(/* some work */);
});
```
The `handle_clone.block_on()` call will block thread B until it's finished, but what I'm not sure of is whether the `Future` passed to `block_on()` will execute on the main thread or thread B. That is, when you create a `new_current_thread` runtime, does that mean current thread at time of creation of the runtime, or current thread at the time of dispatching work onto the runtime?

In any case, one thing I could do better here is to provide a custom `std::clone::Clone()` impl on `AwsSqsTaskQueue` that would clone every other member of the struct but instantiate a new `Runtime`, so that then at least each instance of `AwsSqsTaskQueue` wouldn't create a new runtime every time it makes an API call.

Reply:
I spoke too soon -- I can't instantiate a new `Runtime` in `std::clone::Clone::clone()` because `basic_runtime()` can fail, and `clone()` must always succeed. Also I ran an experiment with cloned `tokio::runtime::Handle`s and it's not promising:
```
#[test]
    fn test_current_thread() {
        print_current_thread_id("test function");

        let runtime = Builder::new_current_thread().enable_all().build().unwrap();
        runtime.block_on(async {
            print_current_thread_id("block_on runtime");
        });
        runtime.handle().spawn_blocking(|| {
            print_current_thread_id("handle spawn_blocking");
        });
        let handle = runtime.handle().clone();
        handle.spawn_blocking(|| {
            print_current_thread_id("handle clone spawn_blocking");
        });

        thread::spawn(move || {
            print_current_thread_id("thread");
            handle.spawn_blocking(|| {
                print_current_thread_id("spawn blocking from thread");
            });

            let runtime = Builder::new_current_thread().enable_all().build().unwrap();
            runtime.block_on(async {
                print_current_thread_id("block on from runtime created in thread");
            })
        }).join().unwrap();

        assert!(false);
    }

    fn print_current_thread_id(context: &str) {
        println!("current thread ID in {}: {:?}", context, current().id());
    }
```

yields:

```
current thread ID in test function: ThreadId(71)
current thread ID in block_on runtime: ThreadId(71)
current thread ID in handle spawn_blocking: ThreadId(72)
current thread ID in handle clone spawn_blocking: ThreadId(72)
current thread ID in thread: ThreadId(74)
current thread ID in spawn blocking from thread: ThreadId(72)
current thread ID in block on from runtime created in thread: ThreadId(74)
```
So what this tells us is that if we use `tokio::runtime::Builder::new_current_thread` to create a `tokio::runtime::Handle` in thread A, then clone that `Handle` to threads B, C, etc., then work sent to the cloned handles will all execute on thread A. We don't want this, because it would mean every cloned instance of `AwsSqsTaskQueue` would contend for a single thread.

So, to recap:
- we can't clone a `Runtime`
- we _can_ clone a `Handle`, but it won't do what we want to do
- we can't provide a custom `Clone` impl that instantiates a new `Runtime` because `clone()` can't fail

I'm not sure how to do better. We could have a field `Option<Runtime>` on `AwsSqsTaskQueue` and lazily initialize it from any method that needs the runtime, but honestly that seems like more trouble than it's worth unless we know that instantiating a `Runtime` is particularly expensive.